tests/functional/backbeat/listLifecycleCurrents.js

```diff
@@ -551,44 +616,78 @@ describe('listLifecycleCurrents with bucket versioning enabled and delete object
     const expectedKeyVersions = {};
 
     before(done => async.series([
-            next => s3.createBucket({ Bucket: testBucket }, next),
-            next => s3.putBucketVersioning({
+            next => {
+                getCredentials()
+                    .then(creds => {
+                        credentials = creds;
+                        return getS3Hostname();
+                    })
+                    .then(() => next())
+                    .catch(next);
+            },
+            next => s3.send(new CreateBucketCommand({ Bucket: testBucket }))
+                .then(() => next())
+                .catch(next),
+            next => s3.send(new PutBucketVersioningCommand({
                 Bucket: testBucket,
                 VersioningConfiguration: { Status: 'Enabled' },
-            }, next),
-            next => {
-                s3.putObject({ Bucket: testBucket, Key: keyName0, Body: '123', Tagging: 'mykey=myvalue' },
-                (err, data) => {
-                    if (err) {
-                        next(err);
-                    }
+            }))
+                .then(() => next())
+                .catch(next),
+            next => s3.send(new PutObjectCommand({
+                Bucket: testBucket,
+                Key: keyName0,
+                Body: '123',
+                Tagging: 'mykey=myvalue',
+            }))
+                .then(data => {
                     expectedKeyVersions[keyName0] = data.VersionId;
-                    return next();
-                });
-            },
-            next => s3.putObject({ Bucket: testBucket, Key: keyName1, Body: '123', Tagging: 'mykey=myvalue' }, next),
-            next => s3.deleteObject({ Bucket: testBucket, Key: keyName1 }, next),
-            next => s3.putObject({ Bucket: testBucket, Key: keyName2, Body: '123', Tagging: 'mykey=myvalue' },
-                (err, data) => {
-                    if (err) {
-                        next(err);
-                    }
+                    next();
+                })
+                .catch(next),
+            next => s3.send(new PutObjectCommand({
+                Bucket: testBucket,
+                Key: keyName1,
+                Body: '123',
+                Tagging: 'mykey=myvalue',
+            }))
+                .then(() => next())
+                .catch(next),
+            next => s3.send(new DeleteObjectCommand({ Bucket: testBucket, Key: keyName1 }))
```

Comment:
Here too (and it's not the only one)
```suggestion
            async () => s3.send(new DeleteObjectCommand({ Bucket: testBucket, Key: keyName1 })),
```

Reply:
I'd rather keep the next here it's part of a chain